yt/data_objects/selection_data_containers.py

```diff
@@ -871,6 +871,13 @@ def __init__(self, data_source, conditionals, ds=None,
                 raise RuntimeError(
                     "Cannot use both base_object and data_source")
             data_source=base_object
+        if isinstance(data_source, YTCutRegion):
+            self.__init__(
+                data_source.base_object,
+                data_source.conditionals + conditionals,
```

Comment:
Calling `__init__` inside of `__init__` is a little weird to me. Is it possible to set `data_source` and `conditionals` in this if block and then set up the instance using the existing logic below?

Also I realize this is how it is already, but a straightforward cleanup would be to deprecate (with a visible warning) passing `base_object` in and to call the attribute `data_source` instead of `base_object`.

Reply:
Yeah, this can probably be simplified like that. I'll give it a go.